cni/deployments/kubernetes/install/scripts/install-cni.sh

```diff
@@ -25,208 +25,297 @@ set -u -e
-#
 function find_cni_conf_file() {
-    cni_cfg=
+    local cni_cfg=
     for cfgf in "${MOUNTED_CNI_NET_DIR}"/*; do
         if [ "${cfgf: -5}" = ".conf" ]; then
             # check if it's a valid CNI .conf file
-            type=$(jq 'has("type")' < "${cfgf}" 2>/dev/null)
+            local type=$(jq 'has("type")' < "${cfgf}" 2>/dev/null)
             if [ "${type}" = "true" ]; then
                 cni_cfg=$(basename "${cfgf}")
                 break
             fi
         elif [ "${cfgf: -9}" = ".conflist" ]; then
             # Check that the file is json and has top level "name" and "plugins" keys
             # NOTE: "cniVersion" key is not required by libcni (kubelet) to be present
-            name=$(jq 'has("name")' < "${cfgf}" 2>/dev/null)
-            plugins=$(jq 'has("plugins")' < "${cfgf}" 2>/dev/null)
+            local name=$(jq 'has("name")' < "${cfgf}" 2>/dev/null)
+            local plugins=$(jq 'has("plugins")' < "${cfgf}" 2>/dev/null)
             if [ "${name}" = "true" ] && [ "${plugins}" = "true" ]; then
                 cni_cfg=$(basename "${cfgf}")
                 break
             fi
         fi
     done
-    echo "$cni_cfg"
+    echo "${cni_cfg}"
 }
 
-function check_install() {
-  cfgfile_nm=$(find_cni_conf_file)
-  if [ "${cfgfile_nm}" != "${CNI_CONF_NAME}" ]; then
-    if [ -n "${CNI_CONF_NAME_OVERRIDE}" ]; then
-       # Install was run with overridden cni config file so don't error out on the preempt check.
-       # Likely the only use for this is testing this script.
-       echo "WARNING: Configured CNI config file \"${MOUNTED_CNI_NET_DIR}/${CNI_CONF_NAME}\" preempted by \"$cfgfile_nm\"."
-    else
-       echo "ERROR: CNI config file \"${MOUNTED_CNI_NET_DIR}/${CNI_CONF_NAME}\" preempted by \"$cfgfile_nm\"."
-       exit 1
-    fi
+# Initializes required variables
+function init() {
+  echo "Initializing variables"
+  declare -a FILE_CLEANUP_ARR
+
+  # The directory on the host where CNI networks are installed. Defaults to
+  # /etc/cni/net.d, but can be overridden by setting CNI_NET_DIR.  This is used
+  # for populating absolute paths in the CNI network config to assets
+  # which are installed in the CNI network config directory.
+  HOST_CNI_NET_DIR=${CNI_NET_DIR:-/etc/cni/net.d}
+  MOUNTED_CNI_NET_DIR=${MOUNTED_CNI_NET_DIR:-/host/etc/cni/net.d}
+
+  KUBECFG_FILE_NAME=${KUBECFG_FILE_NAME:-ZZZ-istio-cni-kubeconfig}
+  CFGCHECK_INTERVAL=${CFGCHECK_INTERVAL:-1}
+
+  # Whether the Istio CNI plugin should be installed as a chained plugin (defaults to true) or as a standalone plugin (when false)
+  CHAINED_CNI_PLUGIN=${CHAINED_CNI_PLUGIN:-true}
+
+  # if user sets CNI_CONF_NAME, script will use that value
+  CNI_CONF_NAME=${CNI_CONF_NAME:-}
+  # else if a CNI config file exists, default to first CNI config file (will overwrite config)
+  # use internal var to allow loop to restart without modifying user defined env var
+  cni_conf_name=${CNI_CONF_NAME:-$(find_cni_conf_file)}
```

Comment:
`local ...` to be consistent with other functions above?

Reply:
The `cni_conf_name` variable is not local to the function and used throughout the script. I just make it a lower case version of the user defined `CNI_CONF_NAME` env var. This way, any modifications to the `cni_conf_name` variable does not affect the user's configuration.

Reply:
ok